### PR TITLE
Rework of CI files for oadp 1.3, 1.4 and master

### PR DESCRIPTION
**Pull request by mpryc**
Rework of CI files for oadp 1.3, 1.4 and master
    
    The rework includes:
     - Small src fix for use of OCP 4.12 and 4.14 on Branch 1.0
     - Removal of branch 1.1 and 1.2 of oadp-operator
     - Addition of OCP 4.15 on branch 1.3
     - Move branch 1.4 to OCP 4.14, 4.15 and 4.16
     - Move branch master to OCP 4.14, 4.15 and 4.16
     - Removal of src from builder where needed
     - Addition of virt jobs to relevant branches
     - Removal of gcp jobs
     - Removal of azure jobs

### CODE REVIEW

File: ci-operator/config/openshift/oadp-operator/openshift-oadp-operator-master__4.15.yaml [hunk 1]
```diff
@@ -180,8 +180,34 @@ tests:
           cpu: 1000m
           memory: 512Mi
     workflow: optional-operators-ci-azure
+- always_run: false
```
Comment (mateusoliveira43) [inline ci-operator/config/openshift/oadp-operator/openshift-oadp-operator-master__4.15.yaml:183]: also need to remove from 4.14, right?

File: ci-operator/config/openshift/oadp-operator/openshift-oadp-operator-oadp-1.4__4.15.yaml [hunk 2]
```diff
@@ -18,7 +18,7 @@ releases:
   latest:
     release:
       channel: fast
-      version: "4.12"
+      version: "4.15"
```
Comment (mateusoliveira43) [inline ci-operator/config/openshift/oadp-operator/openshift-oadp-operator-oadp-1.4__4.15.yaml:21]: also need to add virt tests in oadp-1.4, right?

File: ci-operator/config/openshift/oadp-operator/openshift-oadp-operator-master__4.15.yaml (hunk 1 above)
Comment (mateusoliveira43) [reply ci-operator/config/openshift/oadp-operator/openshift-oadp-operator-master__4.15.yaml:183]: no need

Comment (mateusoliveira43) [reply ci-operator/config/openshift/oadp-operator/openshift-oadp-operator-master__4.15.yaml:183]: (I can not resolve the comment myself, but it is resolved for me :smile: )

**Review by mateusoliveira43 — APPROVED** [review]

**Review by mateusoliveira43 — APPROVED** [review]

**Comment by mateusoliveira43** [comment]
/pj-rehearse pull-ci-openshift-oadp-operator-oadp-1.4-4.15-e2e-test-kubevirt-aws

**Review by sseago — APPROVED** [review]

**Review by shubham-pampattiwar — APPROVED** [review]

**Comment by mpryc** [comment]
/pj-rehearse pull-ci-openshift-oadp-operator-oadp-1.4-4.15-e2e-test-kubevirt-aws

File: ci-operator/config/openshift/oadp-operator/openshift-oadp-operator-oadp-1.0__4.7.yaml [hunk 3]
```diff
@@ -5,10 +5,8 @@ build_root:
     tag: rhel-8-golang-1.19-openshift-4.12
 images:
 - dockerfile_path: Dockerfile
-  from: src
```
Comment (mateusoliveira43) [inline ci-operator/config/openshift/oadp-operator/openshift-oadp-operator-oadp-1.0__4.7.yaml:8]: I would undo these changes to non master branch files

I believe the problem does not happen in those branches

**Comment by mpryc** [comment]
rebased still need to address src request from @mateusoliveira43

File: ci-operator/config/openshift/oadp-operator/openshift-oadp-operator-oadp-1.0__4.7.yaml (hunk 3 above)
Comment (mpryc) [reply ci-operator/config/openshift/oadp-operator/openshift-oadp-operator-oadp-1.0__4.7.yaml:8]: As discussed offline removing this line won't break, so we will have this in sync between different builders/branches.

File: ci-operator/config/openshift/oadp-operator/openshift-oadp-operator-oadp-1.3__4.12.yaml [hunk 4]
```diff
@@ -2,13 +2,11 @@ build_root:
   image_stream_tag:
     name: builder
     namespace: ocp
-    tag: rhel-8-golang-1.20-openshift-4.14
+    tag: rhel-8-golang-1.19-openshift-4.12
```
Comment (mateusoliveira43) [inline ci-operator/config/openshift/oadp-operator/openshift-oadp-operator-oadp-1.3__4.12.yaml:5]: I would undo these changes of image in 1.3 branch, it goes against our standards https://github.com/openshift/oadp-operator/blob/master/docs/developer/PROW_CI.md?plain=1#L73-L81

i.e., 4.12 and 4.13 jobs would use Go 1.19, but 4.14 and 4.15 jobs use Go 1.20

Comment (mpryc) [reply ci-operator/config/openshift/oadp-operator/openshift-oadp-operator-oadp-1.3__4.12.yaml:5]: why 4.12 and 4.13 should not use go 1.20 ? what about oadp 1.0 and branch 4.14?

I think having per ocp image makes sense as we are testing as well variety of go versions almost for free here.

**Review by mateusoliveira43 — APPROVED** [review]
lets get ball rolling

update doc!

some failures are changes that need to be done in branches (like pining version of tool or removing restartPolicy from templates)

/lgtm

**Comment by mpryc** [comment]
/pj-rehearse periodic-ci-openshift-oadp-operator-oadp-1.0-unit-test-periodic

**Comment by mpryc** [comment]
/pj-rehearse pull-ci-openshift-oadp-operator-oadp-1.4-unit-test

**Comment by kaovilai** [comment]
Looks like 1.0 branch missing envtest non-latest update.

**Comment by kaovilai** [comment]
https://github.com/openshift/velero/pull/321

**Comment by kaovilai** [comment]
https://github.com/openshift/oadp-operator/blob/oadp-1.0/Makefile#L222 is still pointing to latest.

**Comment by mpryc** [comment]
> https://github.com/openshift/oadp-operator/blob/oadp-1.0/Makefile#L222 is still pointing to latest.

https://github.com/openshift/oadp-operator/pull/1428

**Review by weshayutin — APPROVED** [review]
/LGTM

**Comment by mateusoliveira43** [comment]
/pj-rehearse pull-ci-openshift-oadp-operator-oadp-1.4-unit-test pull-ci-openshift-oadp-operator-oadp-1.3-unit-test pull-ci-openshift-oadp-operator-oadp-1.0-unit-test pull-ci-openshift-oadp-operator-master-unit-test

**Comment by mpryc** [comment]
/pj-rehearse periodic-ci-openshift-oadp-operator-oadp-1.0-unit-test-periodic
